src/main/java/net/minecraftforge/datafix/ForgeDataFixerCustomFixersModStateTransition.java

```diff
@@ -0,0 +1,220 @@
+    }
+
+    /**
+     * Builder function for creating the events that are fired on start up.
+     * This returns one generator per schema, and keeps the schema in the order defined.
+     *
+     * The returned generator creates one event internally per schema and mod container, and then stores them for reuse.
+     *
+     * @return The stream of event generators used to fire one event per mod per schema.
+     */
+    private static Stream<IModStateTransition.EventGenerator<?>> generateFixersEvents()
+    {
+        //Grab the schemas
+        List<ForgeSchema> schemas = Lists.newArrayList(getAllSchemas());
+        //Sort them based on their version key.
+        schemas.sort(Comparator.comparingInt(Schema::getVersionKey));
+
+        //Create one generator per schema.
+        return schemas.stream().map(InternalEventGenerator::new);
+    }
+
+    /**
+     * Invoked by the event management system to indicate that we are about the fire an event for the given generator to all mods.
+     * This is invoked exactly once, and only once, per generator (and as such per event, and schema).
+     *
+     * Clears out the current status of the schema by resetting its type definition maps.
+     *
+     * @param executor The executor to execute this task on.
+     * @param eventGenerator The event generator that is about to be executed.
+     * @return A future that indicates success or failure.
+     */
+    private static CompletableFuture<List<Throwable>> preDispatchFixerEvents(
+      final Executor executor, final IModStateTransition.EventGenerator<RegisterFixesEvent> eventGenerator
+    )
+    {
+        if (!(eventGenerator instanceof InternalEventGenerator internalEventGenerator)) {
+            return CompletableFuture.completedFuture(Collections.emptyList());
+        }
+
+        return CompletableFuture.runAsync(() -> {
+            //Notify of the update that is about to start.
+            LOGGER.debug("Resetting and collecting custom fixes in DFU for: {}", internalEventGenerator.schema.getVersionKey());
+
+            //And reset the stored fixes.
+            internalEventGenerator.schema.resetCustomFixes();
+        }, executor).thenApply(v -> Collections.emptyList());
+    }
+
+    /**
+     * Invoked by the event management system to indicate that it has fired all the events for the given schema.
+     * This allows us to collect the data from the events and trigger the given schema to rebuild its type map.
+     *
+     * @param executor The executor to execute the task on.
+     * @param eventGenerator The event generator for the event (and thus the schema in question).
+     * @return The future that indicates success or failure.
+     */
+    private static CompletableFuture<List<Throwable>> postDispatchFixerEvents(
+      final Executor executor, final IModStateTransition.EventGenerator<RegisterFixesEvent> eventGenerator
+    ) {
+        if (!(eventGenerator instanceof InternalEventGenerator internalEventGenerator)) {
```

Comment:
Formatting: braces on newlines.
```suggestion
    )
    {
        if (!(eventGenerator instanceof InternalEventGenerator internalEventGenerator))
        {
```